### PR TITLE
Test: `response correction` module

### PR DESCRIPTION
**Pull request by marki1an**
### 🔧 Type of changes
- [ ] new bid adapter
- [ ] update bid adapter
- [ ] new feature
- [ ] new analytics adapter
- [ ] new module
- [ ] bugfix
- [ ] documentation
- [ ] configuration
- [x] tech debt (test coverage, refactorings, etc.)

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 🏎 Quality check

- [ ] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+
+        and: "Default bid request with APP and Video imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].setAdm(null)
+            seatbid[0].bid[0].nurl = PBSUtils.randomString
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:283]: What purpose of `nurl` in this test?

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+        def start = Instant.now()
+
+        and: "Default bid request with APP and Video imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].setAdm(PBSUtils.getRandomCase("<${PBSUtils.randomString}VAST${PBSUtils.randomString}"))
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:320]: `<` is typo or required?

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+    }
+
+    def "PBs shouldn't modify response meta.mediaType to video and emit logs when requested impression with video and adm obj with asset"() {
+        given: "Start up time"
+        def start = Instant.now()
+
+        and: "Default bid request with APP and audio imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:437]: Why not `getDefaultVideoRequest(APP)`?

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+        assert !response?.seatbid[0]?.bid[0]?.ext?.prebid?.meta?.mediaType
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+    }
+
+    def "PBs should modify response meta.mediaType to video and type to banner and emit logs when requested impression with native and adm value is not asset"() {
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:470]: Maybe: `PBS should modify meta.mediaType and type for original response and also emit logs when response contains native meta.mediaType and adm without asset`

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+        and: "Response should contain seatBid type and mediaType"
+        assert response.seatbid.size() == 1
+        assert response.seatbid[0].bid[0].ext.prebid.type == VIDEO
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+    }
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:428]: Duplicate

**Review by osulzhenko — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy (hunk 1 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:283]: ResponseBidValidator:131L -> validateTypeSpecific(...)

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy (hunk 2 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:320]: Required

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+
+    def "PBS shouldn't modify response with adm obj when request includes #distributionChannel distribution channel"() {
+        given: "Start up time"
+        def start = Instant.now()
+
+        and: "Default bid request video imp"
+        def bidRequest = getDefaultBidRequest(distributionChannel).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:90]: can be replaced with `getDefaultVideoRequest`. Same for others

File: src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy [hunk 7]
```diff
@@ -0,0 +1,507 @@
...
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithResponseCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "PBS shouldn't emit log"
+        def logsByTime = pbsServiceWithResponseCorrectionModule.getLogsByTime(start)
+        assert getLogsByText(logsByTime, bidResponse.seatbid[0].bid[0].id).size() == 0
+
+        and: "Response should contain seatBid and video media type"
+        assert response.seatbid.size() == 1
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy:65]: But we don't check any `video media type`

**Review by osulzhenko — CHANGES_REQUESTED** [review]

**Review by osulzhenko — APPROVED** [review]

**Pull request merged**